packages/playwright-test/src/runner.ts

```diff
@@ -243,6 +243,13 @@ export class Runner {
         return true;
       const name = project.name.toLocaleLowerCase();
       unknownProjects!.delete(name);
+      for (const value of projectsToFind) {
+        const match = new RegExp('^' + value.replace(/\*/g, '.*').replace(/\?/g, '.?') + '$');
```

Comment:
How do you feel about using regex here from the get go? The users would need to put `foo.*` to include all projects with prefix foo, but they get rich matching consistent with our test filter and `--grep` treatment. There is a `forceRegExp` method that is used for command like args that are considered regex.

Reply:
Also, I don't quite understand what the purpose of `forceRegExp` is. Won't that require users to do `--project /foo.*/`?

Reply:
> Should I still do ^ and $? Is it okay for --project r to match chromium and firefox?

Oh, that's a good point - we don't want to run `non-basic` when `--project basic` is passed.

> Won't that require users to do --project /foo.*/?

I don't think it will, the way I read it is that it'll drop surrounding slashes if they were provided while constructing the regex

Reply:
> Oh, that's a good point - we don't want to run `non-basic` when `--project basic` is passed.

`const match = forceRegExp('/^' + value + '$/i');`
Would this be an acceptable solution?

Reply:
Yes, you might want to strip the surrounding // if they are present, other than that it looks fair.

Reply:
I think that making the user use \ if their project is `/foo/` is worse than them having to remove their // if they try to do a normal regex search.

Reply:
I'm not sure I follow this

Reply:
If you strip surrounding //, then if a user has a project that's called `/foo/` they won't be able to find it unless they do `--project //foo//` or `--project \/foo\/`.

If you don't strip them, then if a user is trying to use `--project /foo/` to find project `foo`, they'll have to remove the `//` to find it.

Both cases are unlikely, but I didn't strip them because I think the second case is easier for the user to handle. I might be wrong, though.